plaid/plaid.go

```diff
@@ -30,39 +29,38 @@ type Client struct {
 	httpClient  *http.Client
 }
 
+type ClientOptions struct {
+	ClientID    string
+	Secret      string
+	PublicKey   string
+	Environment PlaidEnvironment
+	HttpClient  *http.Client
+}
+
 // NewClient instantiates a Client associated with a client id, secret and environment.
-func NewClient(clientID, secret, publicKey string, environment PlaidEnvironment) *Client {
-	if !environment.IsValid() {
-		fmt.Println(errors.New("Invalid environment specified: " + string(environment)))
-		return nil
+func NewClient(options ClientOptions) (client *Client, err error) {
+	if !options.Environment.Valid() {
+		return client, errors.New("Invalid environment specified: " + string(options.Environment))
 	}
 
-	return &Client{clientID, secret, publicKey, environment, &http.Client{}}
-}
-
-// NewCustomClient is the same as NewClient but with an additional http.Client parameter.
-// This is required if you want to run the code on App Engine, which
-// prohibits the use of http.DefaultClient
-func NewCustomClient(clientID, secret, publicKey string, environment PlaidEnvironment, httpClient *http.Client) *Client {
-	if !environment.IsValid() {
-		fmt.Println(errors.New("Invalid environment specified: " + string(environment)))
-		return nil
+	if options.HttpClient == nil {
+		options.HttpClient = &http.Client{}
 	}
 
-	return &Client{clientID, secret, publicKey, environment, httpClient}
+	return &Client{options.ClientID, options.Secret, options.PublicKey, options.Environment, options.HttpClient}, nil
```

Comment:
nit: I would prefer the full named object instantaition for the same reason we introduced `ClientOptions` (with the current approach, you have to consult the type to determine the proper order to pass the options into the Client struct).

e.g. 

```go
return &Client{
  clientID: options.ClientID,
  secret: options.Secret,
  // etc...
}
```

Reply:
⚡️